torch/_inductor/utils.py

```diff
@@ -2402,17 +2507,17 @@ def boolean_ops():
 @dataclasses.dataclass
 class OpDtypeRule:
     type_promotion_kind: ELEMENTWISE_TYPE_PROMOTION_KIND
-    override_return_dtype: Optional[torch.dtype]
+    override_return_dtype: torch.dtype | None
```

Comment:
is this `torch.dtype | None` canonical ? could we do it in the whole codebase as a single pr. id rather avoid churn and local inconsistency. 

cc @aorenste @bobrenjc93

Reply:
Happy to embrace this as a change but would prefer to do it as a larger code mod instead of inconsistency inside the codebase. see https://github.com/pytorch/pytorch/pull/144528